implementations/rust/ockam/ockam/src/credential/exchange.rs

```diff
@@ -64,68 +191,80 @@ impl CredentialExchange {
             )),
         }
     }
+
+    async fn present_credential_mutual(
+        &self,
+        credential: Credential<'_>,
+        receiver: &IdentityIdentifier,
+        route: impl Into<Route>,
+        authorities: &BTreeMap<IdentityIdentifier, IdentityChangeHistory>,
+        authenticated_storage: &impl AuthenticatedStorage,
+        vault: &impl IdentityVault,
+    ) -> Result<()> {
+        let mut child_ctx = self.ctx.new_detached(Address::random_local()).await?;
+        let path = "actions/present_mutual";
+        let buf = request(
+            &mut child_ctx,
+            "credential",
+            None,
+            route.into(),
+            Request::post(path).body(credential),
+        )
+        .await?;
+
+        let mut dec = Decoder::new(&buf);
+        let res: Response = dec.decode()?;
+        match res.status() {
+            Some(Status::Ok) => {}
+            _ => {
+                return Err(ockam_core::Error::new(
+                    Origin::Application,
+                    Kind::Invalid,
+                    "credential presentation failed",
+                ))
+            }
+        }
+
+        let credential: Credential = dec.decode()?;
+
+        Self::receive_presented_credential(
```

Comment:
Should this not act on error?

Reply:
What do you mean?

Reply:
`receive_presented_credential` may fail to validate the credential and returns an error response in this case which is ignored here. I had some difficulties following the code so I may be missing something.

Reply:
Nice catch. I'll fix, but this requires refactor in the future :D